### PR TITLE
Modified label used to run cuda jobs

### LINKED ISSUE #3000: Rename label `ci: ready to be merged`
Label `ci: ready to be merged` is used to trigger CUDA builds.
Let's rename it `ci: run CUDA builds` accordingly.

### PR DESCRIPTION
**Pull request by CusiniM**
I think it's time for us to do this. 

I have left the old label for now. We can delete it once we merge this.

### CODE REVIEW

**Review by ryar9534 — APPROVED** [review]

**Review by TotoGaz — APPROVED** [review]
You can change the name of the label through the web ui when you merge this.

**Comment by CusiniM** [comment]
do we prefer `CUDA jobs` or `CUDA builds` ?

**Comment by CusiniM** [comment]
> You can change the name of the label through the web ui when you merge this.

yeah but there s 2 labels now. I can just ditch the old one.

**Comment by ryar9534** [comment]
> do we prefer `CUDA jobs` or `CUDA builds` ?

I personally prefer `CUDA builds`

**Comment by TotoGaz** [comment]
> > You can change the name of the label through the web ui when you merge this.
> 
> yeah but there s 2 labels now. I can just ditch the old one.

Behind the scenes, I guess that labels have an GitHub internal identifier.
Changing the name of the label will automatically spread to all the PRs (past, present and future) with no further action required. Up to you.

**Comment by CusiniM** [comment]
> > > You can change the name of the label through the web ui when you merge this.
> > 
> > 
> > yeah but there s 2 labels now. I can just ditch the old one.
> 
> Behind the scenes, I guess that labels have an GitHub internal identifier. Changing the name of the label will automatically spread to all the PRs (past, present and future) with no further action required. Up to you.

yes, if we care about past prs yes. For the present ones it would actually be good to force ppl to make the change manually so that they notice it.

File: .github/workflows/ci_tests.yml [hunk 1]
```diff
@@ -320,7 +320,7 @@ jobs:
       HOST_CONFIG: ${{ matrix.HOST_CONFIG }}
       NPROC: ${{ matrix.NPROC }}
       RUNS_ON: ${{ matrix.RUNS_ON }}
-      REQUIRED_LABEL: "ci: ready to be merged"
+      REQUIRED_LABEL: "ci: run CUDA jobs"
```
Comment (CusiniM) [inline .github/workflows/ci_tests.yml]:
```suggestion
      REQUIRED_LABEL: "ci: run CUDA builds"
```

File: .github/workflows/ci_tests.yml [hunk 2]
```diff
@@ -238,7 +238,7 @@ jobs:
       GCP_BUCKET: geosx/ubuntu22.04-gcc11
       RUNS_ON: Runner_4core_16GB
 
-  # If the 'ci: ready to be merged' PR label is found, the cuda jobs run immediately along side linux jobs.
+  # If the 'ci: run CUDA jobs' PR label is found, the cuda jobs run immediately along side linux jobs.
```
Comment (CusiniM) [inline .github/workflows/ci_tests.yml]:
```suggestion
  # If the 'ci: run CUDA builds' PR label is found, the cuda jobs run immediately along side linux jobs.
```

**Review by rrsettgast — APPROVED** [review]

**Pull request merged**